cmd/controller/controller.go

```diff
@@ -224,6 +224,14 @@ func (c *command) start(ctx context.Context) error {
 		nodeComponents.Add(ctx, controllerLeaseCounter)
 	}
 
+	enableExternalLB := !c.SingleNode && !slices.Contains(c.DisableComponents, constant.ExternalLBComponentName)
```

Comment:
This is added before apiserver as the VIP may be a requirement for the apiserver depending on the configuration (for instance in the inttest). Hence I don't think it's a good idea to allow dynamic configuration changes for this.

There is no technical reason why we cannot, however I think we shouldn't.